exchanges/deribit/deribit_websocket_eps.go

```diff
@@ -0,0 +1,2139 @@
+	if err != nil {
+		return "", err
+	}
+	if resp != "ok" {
+		return "", fmt.Errorf("setting announcement %v as read failed", id)
+	}
+	return resp, nil
+}
+
+// WSSetEmailForSubAccount links an email given to the designated subaccount through the websocket connection.
+func (d *Deribit) WSSetEmailForSubAccount(sid int64, email string) (string, error) {
+	if sid <= 0 {
+		return "", fmt.Errorf("%w, invalid subaccount user id", errInvalidID)
+	}
+	if !common.MatchesEmailPattern(email) {
+		return "", errInvalidEmailAddress
+	}
+	input := &struct {
+		SID   int64  `json:"sid"`
+		Email string `json:"email"`
+	}{
+		Email: email,
+		SID:   sid,
+	}
+	var resp interface{}
+	err := d.SendWSRequest(setEmailForSubAccount, input, &resp, true)
+	if err != nil {
+		return "", err
+	}
+	_, ok := resp.(map[string]interface{})
+	if ok {
+		data, err := json.Marshal(resp)
+		if err != nil {
+			return "", err
+		}
+		var respo TFAChallenge
+		err = json.Unmarshal(data, &respo)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	}
+	if resp != "ok" {
+		return "", fmt.Errorf("could not link email (%v) to subaccount %v", email, sid)
+	}
+	return "resp", nil
+}
+
+// WSSetEmailLanguage sets a requested language for an email through the websocket connecton.
+func (d *Deribit) WSSetEmailLanguage(language string) (string, error) {
+	if language == "" {
+		return "", errors.New("language is required")
+	}
+	var resp string
+	err := d.SendWSRequest(setEmailLanguage, map[string]string{"language": language}, &resp, true)
+	if err != nil {
+		return "", err
+	}
+	if resp != "ok" {
+		return "", fmt.Errorf("could not set the email language to %v", language)
+	}
+	return resp, nil
+}
+
+// WSSetPasswordForSubAccount sets a password for subaccount usage through the websocket connection.
+func (d *Deribit) WSSetPasswordForSubAccount(sid int64, password string) (interface{}, error) {
+	if sid <= 0 {
+		return "", fmt.Errorf("%w, invalid subaccount user id", errInvalidID)
+	}
+	if password == "" {
+		return "", errors.New("subaccount password must not be empty")
+	}
+	input := &struct {
+		Password string `json:"password"`
+		SID      int64  `json:"sid"`
+	}{
+		Password: password,
+		SID:      sid,
+	}
+	var resp interface{}
+	err := d.SendWSRequest(setPasswordForSubAccount, input, &resp, true)
+	if err != nil {
+		return "", err
+	}
+	_, ok := resp.(map[string]interface{})
+	if ok {
+		data, err := json.Marshal(resp)
+		if err != nil {
+			return "", err
+		}
+		var respo TFAChallenge
+		err = json.Unmarshal(data, &respo)
+		if err != nil {
+			return nil, err
+		}
+		return respo, nil
+	}
+	if resp != "ok" {
+		return "", fmt.Errorf("could not set the provided password to subaccount %v", sid)
+	}
+	return "ok", nil
+}
+
+// WSToggleNotificationsFromSubAccount toggles the notifications from a subaccount specified through the websocket connection.
+func (d *Deribit) WSToggleNotificationsFromSubAccount(sid int64, state bool) (string, error) {
```

Comment:
Can you  go over and check the string returns, I don't know if its necessary to return them as they are checked below.